vm/package/deploy.rs

```diff
@@ -71,39 +78,47 @@ impl<'de, N: Network> Deserialize<'de> for DeployRequest<N> {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
```

Comment:
We should also check that the deserialized transaction here is a `Transaction::Deploy`